src/System.Net.Http/src/System/Net/Http/StreamToStreamCopy.cs

```diff
@@ -8,69 +8,52 @@
 namespace System.Net.Http
 {
     // This helper class is used to copy the content of a source stream to a destination stream.
-    // The type verifies if the source and/or destination stream are MemoryStreams (or derived types). If so, sync 
+    // The type verifies if the source and/or destination stream are MemoryStreams (or derived types). If so, sync
     // read/write is used on the MemoryStream to avoid context switches.
-    internal class StreamToStreamCopy
+    internal static class StreamToStreamCopy
     {
-        private byte[] _buffer;
-        private int _bufferSize;
-        private Stream _source;
-        private Stream _destination;
-        private bool _disposeSource;
-        private bool _sourceIsMemoryStream;
-        private bool _destinationIsMemoryStream;
-
-        public StreamToStreamCopy(Stream source, Stream destination, int bufferSize, bool disposeSource)
+        public static async Task CopyAsync(Stream source, Stream destination, int bufferSize, bool disposeSource)
         {
             Contract.Requires(source != null);
             Contract.Requires(destination != null);
             Contract.Requires(bufferSize > 0);
 
-            _buffer = new byte[bufferSize];
-            _source = source;
-            _destination = destination;
-            _bufferSize = bufferSize;
-            _disposeSource = disposeSource;
-            _sourceIsMemoryStream = source is MemoryStream;
-            _destinationIsMemoryStream = destination is MemoryStream;
-        }
+            byte[] buffer = new byte[bufferSize];
 
-        public async Task StartAsync()
-        {
             // If both streams are MemoryStreams, just copy the whole content at once to avoid context switches.
             // This will not block since it will just result in a memcopy.
-            if (_sourceIsMemoryStream && _destinationIsMemoryStream)
+            if (source is MemoryStream && destination is MemoryStream)
```

Comment:
FWIW, this optimization around MemoryStream isn't accomplishing all that much, as MemoryStream overrides ReadAsync and WriteAsync to be synchronous.  Plus, if we really wanted to optimize for MemoryStream, there are additional things that can be done. Ideally I'd think CopyToAsync would be used instead of this helper or at least used by the helper; that way the copying can be done in an optimal manner based on whatever stream type is being used.  If we don't want to use CopyToAsync in the general case because we're concerned about compat, we could use it just in the case where we test to see if the source is a MemoryStream.  And if we still don't want to do that, we could at least avoid allocating the byte[] buffer in the case of a source MemoryStream if we can get its buffer via TryGetBuffer and then just do a single write of that whole buffer to the destination, rather than copying it in chunks through an intermediate buffer we allocate.

Anyway, I realize the changes to this method are really just cosmetic (beyond avoiding the allocation of the containing class), so these comments are more about the existing implementation rather than about the change.